source/storage-item.test-d.ts

```diff
@@ -25,9 +28,22 @@ expectType<Promise<string>>(stringDefault.get());
 expectType<Promise<void>>(stringDefault.set('some string'));
 
 // NonNullable from default, includes broader type as generic
-const broadGeneric = new StorageItem<Record<string, number>>('key', {defaultValue: {}});
+// The second type parameter must be re-specified because TypeScript stops inferring it
+// https://github.com/microsoft/TypeScript/issues/26242
+const broadGeneric = new StorageItem<Record<string, number>, Record<string, number>>('key', {defaultValue: {a: 1}});
```

Comment:
Either there's a regression or this never actually worked. Considering that the linked issue says that partial generic parameters break inference, I'm not sure why this test passed before. 🤷‍♂️